### ISSUE TITLE
Display “Contents Preview” files on the project contents page

### ISSUE THREAD

**Issue by bess** [pulibrary/tiger-data-app#389]
Display “Contents Preview” files on the project contents page
Hector says “the first 100 files are easy” 

![Screenshot 2024-01-24 at 11 28 18 AM](https://github.com/pulibrary/tiger-data-app/assets/65608/863f406e-3719-4f4a-b08f-1ca1653d5d3d)

- Question regarding the asset nodes: Top 100 directories or top 5 directories with top 100 files? Whichever is easiest and whatever looks the most like a traditional file-system (top directories with children). End-users need to be able to readily identify and confirm that this is the project belonging to the end-user. Try and use the default number of files returned within the Mediaflux server response.

- aterm `asset.query` should be used by the assignee, and there are established patterns for this within the code base.

- How are we going to connect a Project to Mediaflux? Querying Mediaflux using the directory name? Querying using the DOI?
- The directory name may not match the namespace. Mediaflux also offers different mount points for different protocols (hence, directory path will be inconsistent). Collection asset ID or Project ID (which is the DOI) should be used. Performance might not be the same for both.
- For `asset.query` commands, a collection asset ID can be used to refine the query for this feature.
- https://github.com/pulibrary/tiger-data-app/blob/main/app/models/mediaflux/http/query_request.rb
- Use the Collection Asset ID for querying the Project. There will need to be documented the namespace for each Project, and then under that one will find the Collection Asset. [PROJECT_NAME] + "-ns", however, names are metadata which can vary; IDs are persistent. Hence, an ADR might need to be created in order to formalize the guidelines for this.
- The model will have the Collection ID
- A hard-coded Collection ID will be used for demonstrating this feature
- Long-term, want to attempt a metadata query for the DOI, but this is a part of another ticket
- This is favored for the long-term solution by everyone, as the Collection Asset ID is arbitrary

- This should retrieve 10-20 assets from Mediaflux
  - This number can be adjusted based upon user feedback
- Which is easier for the preview? Scrolling or preview?
- End-user clicks "Show More" with up to 100 (this is practical for Mediaflux)
  - This expands the "Content Preview" window
- This could be implemented using DataTables
  - DataTables permits one to select whether or not 10-100 assets are rendered
- In terms of performance, this assumes that there will be a noticeable performance impact when end-users query for 10 vs. 100 assets
  - We can start with 100 (this is the Mediaflux default), and this has been consistently fast
  - Then, query by batches of 100 for scrolling or pagination
  - *We should implement the minimum (100) assets in the query response*
    - Only after integrating with the live Mediaflux server will we know how to benchmark
  - When you click "Show More", retrieve 101-200 assets (if there are that many)
  - Only permit "Show More" to be clicked *once*
  - Then we can determine if multiple click opportunities are possible to support later
- #390 should be scoped to address Mediaflux pagination
- DataTables default size per page is 10, but this can be set to 25 or 100
- 200ms is the avg. blink of an eye, this should be the case for the initial page load

Acceptance Criteria:
- [ ] Extend the "Contents Preview" View
- [ ] Integrate DataTables
- [ ] Display whatever Mediaflux gives the client by default when you query by the Collection ID which may be input by hand
- [ ] Document explicitly what Mediaflux gives the client by default (comment on this ticket would be fine)
- [ ] Maintain the "flat" file-directory tree hierarchy implemented in PDC (improvements to this can be introduced later)

**Comment by jrgriffiniii** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/RDSS-61a4f1a12a399b001730f65a/issues/pulibrary/tiger-data-app/389?planning-poker) with Zenhub @bess @carolyncole @hectorcorrea @JaymeeH @jrgriffiniii @kelynch @leefaisonr
